src/OpenSpaceToolkit/Physics/Environment.cpp

```diff
@@ -28,16 +34,39 @@ Environment::Environment(const Instant& anInstant, const Array<Shared<Object>>&
     }
 }
 
+Environment::Environment(
+    const Shared<Object>& aCentralBody, const Array<Shared<Object>>& anObjectArray, const Instant& anInstant
+)
+    : instant_(anInstant),
+      objects_(Array<Shared<const Object>>::Empty()),
+      centralCelestial_(nullptr)
+{
+    objects_.reserve(anObjectArray.getSize() + 1);
+
+    const Shared<const Object> centralBody = Shared<const Object>(aCentralBody->clone());
+    centralCelestial_ = centralBody;
+
+    objects_.add(centralBody);
+    for (const auto& objectSPtr : anObjectArray)
+    {
+        objects_.add(Shared<const Object>(objectSPtr->clone()));
```

Comment:
should we check and raise an error if `centralBody` is also part of the `anObjectArray` ? (to avoid adding it twice)

Reply:
In this case we actually want to store a pointer to the central object so we can easily access it. Otherwise everywhere we would have to loop through the objects + central body.
In this case, we store a shared pointer to the central body so it's lightweight.